### PR TITLE
[av][Android] Bump `com.google.android.exoplayer:exoplayer` from `2.9.2` to `2.13.3`

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Resolves [ENG-2858](https://linear.app/expo/issue/ENG-2858/remove-jcenter)
Continues https://github.com/expo/expo/pull/16122

# How

- Removed `https://github.com/yqritc/Android-ScalableVideoView` by copying the parts into the `expo-av` codebase as it's only available on `jcenter()`
- Bumped the `exoplayer` to the smallest version available on google maven.
- Tweaked the codebase to work with the new version of exoplayer.

# Test Plan

I've succeeded in compiling `bare-expo` and running the `ncl`#`video` screen with both remote and local videos playing nicely.

# TODO

- [x] Ensure `Expo Go` works
- [ ] Backport changes if necessary

### CODE REVIEW

File: android/expoview/build.gradle [hunk 1]
```diff
@@ -99,6 +98,7 @@ repositories {
 
 android {
   compileSdkVersion safeExtGet("compileSdkVersion", 30)
+  ndkVersion "23.0.7421159"
```
Comment (Kudo) [inline android/expoview/build.gradle:101]: any reason we should upgrade ndk version here?

**Review by Kudo — APPROVED** [review]
LGTM. i think android client test error is from expo go, maybe you will update in this pr or another pr.

File: android/expoview/build.gradle (hunk 1 above)
Comment (nomi9995) [reply android/expoview/build.gradle:101]: yes, no need this line, i have checked it on my local

**Comment by mnightingale** [comment]
I've not checked but looking at the release notes for ExoPlayer this possibly resolves a few open av issues (#16139 and maybe #16439).

Is there any reason not to go for a more recent ExoPlayer version (latest is currently 2.17.0, about 10 months more recent than 2.13.3), this PR already makes pretty much all of the required API changes from the last few years so it's not much more work.

A few changes would be required:

- buildMediaSource handle TYPE_RTSP
- A few renamed classes or constants
- Likely upgrade okhttp + okhttp-urlconnection so it uses the same version as ExoPlayer
- A few changes to SharedCookiesDataSourceFactory and CustomHeadersOkHttpDataSourceFactory - might also be able to simplify those using okhttp interceptor, else I think apis are exposed so setDefaultRequestProperties can be used

There are also usages of a few deprecated methods that could be cleaned up.

File: packages/expo-av/android/build.gradle [hunk 2]
```diff
@@ -74,11 +74,9 @@ dependencies {
 
   // Newer version introduces dependency versions conflict
   // on 'com.android.support:support-annotations'
```
Comment (mnightingale) [inline packages/expo-av/android/build.gradle:76]:
```suggestion
```

It appears there isn't a conflict anymore, hence this PR is upgrading exoplayer.

**Comment by dantxal** [comment]
Hello bbarthec!! 

I've been trying to test your solution, I tried to build by linking the dependencies of my repo to your branch on github, but that didn't work.

Now I see that you merged it, could you give me the instructions to get it running on my repo?

This is the repo I've been trying to test it on: [dantxal/ExpoAvIssue](https://github.com/dantxal/ExpoAvIssue).

This is the issue I talked about, for reference: #16323 

Thanks for the good work you've been doing on expo!

**Comment by bbarthec** [comment]
@dantxal, this problem you're reporting is very generic and I have no way to help you with it 😞 
I'm going to publish this change to the npm with the `next` tag and you'll be able to test it that way 👀

**Comment by bbarthec** [comment]
I've just published [`expo-av@11.0.0`](https://www.npmjs.com/package/expo-av/v/11.0.0?activeTab=versions) with the `next` tag on `npm` containing this PR and many more (see [CHANGELOG](https://github.com/expo/expo/blob/main/packages/expo-av/CHANGELOG.md#1100--2022-03-03)) 🎉 
Please try it and feel free to provide any kind of feedback you want (including new issues, because something is broken out of the blue) 😁

**Comment by dantxal** [comment]
@bbarthec You're right, it didn't solve my problem.

To anyone else trying to run `expo-av@11.0.0`: 
You might run into this error on expo package: 
```sh
Module was compiled with an incompatible version of Kotlin.
```

All I had to do to get it running was:

1. Navigate into expo package within node_modules (relative path: `node_modules/expo/android/build.gradle`);
2. Change line 24 from 
```java
    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${safeExtGet('kotlinVersion', '1.4.1')}")
```
into
```java
    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${safeExtGet('kotlinVersion', '1.6.0')}")
```
3. Run `yarn android` again. ( I ran `cd android && ./gradlew clean && cd ..` just to be safe) .

**Comment by eeshankeni** [comment]
Please consider implementing https://github.com/expo/expo/pull/16455 to expo-av 11 as well

**Comment by hirbod** [comment]
So far, expo-av 11 is working fine for me on Android. What still is so extremely annoying is the massive pauseAsync delay on iOS and the framedrops caused by it (from 60 to 20). Android does not have this issues.

**Comment by KrisLau** [comment]
@bbarthec @Kudo Thanks for all the awesome work! I've been working on a fatal error I've been getting on my project and realized that the `exoplayer` version being too low is causing a fatal error when used with `react-native-track-player` which uses v2.18.1. Are there any plans to upgrade to that version any time soon or ado you guys have a workaround for that?

Issue with repro: https://github.com/expo/expo/issues/18937

**Pull request merged**